pilot/pkg/bootstrap/server.go

```diff
@@ -935,6 +936,21 @@ func (s *Server) getCertKeyPaths(tlsOptions TLSOptions) (string, string) {
 
 // setPeerCertVerifier sets up a SPIFFE certificate verifier with the current istiod configuration.
 func (s *Server) setPeerCertVerifier(tlsOptions TLSOptions) error {
+	meshConfig := s.environment.Mesh()
+	caProvider := ""
+	if meshConfig != nil && meshConfig.DefaultConfig != nil &&
+		len(meshConfig.DefaultConfig.ProxyMetadata) > 0 {
+		caProvider = meshConfig.DefaultConfig.ProxyMetadata["CA_PROVIDER"]
+	}
+
+	// Read SpiffeBundleEndpoints from metadata server if not set
+	if features.SpiffeBundleEndpoints == "" && caProvider == "GoogleCA" && platform.IsGCP() {
+		md := platform.NewGCP().Metadata()
+		if projectID, found := md[platform.GCPProject]; found {
+			features.SpiffeBundleEndpoints = projectID + ".svc.id.goog|https://storage.googleapis.com/mesh-ca-resources/spiffe_bundle.json"
```

Comment:
Please don't modify global variables based on dynamically fetching information.. this is going to be a nightmare to debug